troposphere/__init__.py

```diff
@@ -439,9 +439,21 @@ def __init__(self, data):
 
 
 class Tags(AWSHelperFn):
-    def __init__(self, **kwargs):
+    def __init__(self, *args, **kwargs):
+        if not args:
+            # Assume kwargs variant
+            tag_dict = kwargs
+        else:
+            if len(args) != 1:
+                raise(TypeError, "Multiple non-kwargs passed to Tags")
+
+            # Validate single argument passed in is a dict
+            if not isinstance(args[0], dict):
```

Comment:
Maybe use `collections.Mapping` here, in case someone wants to implement something other than `dict`.  Otherwise, this looks good.